docker/test_base/entry_point.sh

```diff
@@ -1,12 +1,20 @@
 #!/bin/bash
-set -eu
+set -eux
 
 TEST_ROOT=/test_root
 OTACLIENT_SRC=/otaclient_src
 OUTPUT_DIR="${OUTPUT_DIR:-/test_result}"
 
-# copy the source code as source is read-only
-cp -R "${OTACLIENT_SRC}" "${TEST_ROOT}"
+mkdir -p ${TEST_ROOT}
+# source code needed to be rw, so copy it to the test_root
+cp -R ${OTACLIENT_SRC}/src ${TEST_ROOT}
+cp ${OTACLIENT_SRC}/uv.lock ${TEST_ROOT}
+# symlink all the other needed folders/files into test root
+ln -s ${OTACLIENT_SRC}/tests ${TEST_ROOT}
+ln -s ${OTACLIENT_SRC}/.git ${TEST_ROOT}
+ln -s ${OTACLIENT_SRC}/pyproject.toml ${TEST_ROOT}
+ln -s ${OTACLIENT_SRC}/README.md ${TEST_ROOT}
+ln -s ${OTACLIENT_SRC}/LICENSE.md ${TEST_ROOT}
```

Comment:
can we move these copies into docker-compose?

Reply:
IMO, we don't need to expose these details into docker-compose file though.

Reply:
OK, maybe I understand.
1. host code is bind to `otaclient_src` as ro in `docker-compose.yml`
```
    - ../..:/otaclient_src:ro
    - ../../test_result:/test_result:rw
```

2. copies the files as rw in `entry_point.sh`
```
cp -R ${OTACLIENT_SRC}/src ${TEST_ROOT}
...
```

This will be to prevent to change host code.
Hmmm, OK, I prefer to aggregate such file copy/bind in a single file. But understand the intention.